tests/test_containerproperties.py

```diff
@@ -400,3 +419,344 @@ def test_SubElementWithSubElementListProperty(self):
+
+    def test_update_xml_value_with_none(self):
+        """Test when the value is None and the property is optional."""
+        self.property._is_optional = True
+        setattr(self.instance, self.property._local_var_name, None)
+        self.property.update_xml_value(self.instance, self.node)
+        self.assertIsNone(self.node.find(self.sub_element_name.text, namespaces=self.node.nsmap))
+
+    def test_update_xml_value_with_values(self):
+        """Test when the value is a list of related QNames."""
+        qname1 = utils.random_qname(namespace=self.sub_element_name.text)
+        qname2 = utils.random_qname(namespace=self.sub_element_name.text)
+        setattr(self.instance, self.property._local_var_name, [qname1, qname2])
+        self.property.update_xml_value(self.instance, self.node)
+        sub_node = self.node.find(self.sub_element_name.text, namespaces=self.node.nsmap)
+        self.assertIsNotNone(sub_node)
+        self.assertEqual(sub_node.text, f'{self.prefix}:{qname1.localname} {self.prefix}:{qname2.localname}')
+
+    def test_update_xml_value_with_random_qnames(self):
+        """Test when the value is a list of random QNames."""
+        qname1 = utils.random_qname()
+        qname2 = utils.random_qname()
+        setattr(self.instance, self.property._local_var_name, [qname1, qname2])
+        self.property.update_xml_value(self.instance, self.node)
+        sub_node = self.node.find(self.sub_element_name.text, namespaces=self.node.nsmap)
+        self.assertIsNotNone(sub_node)
+        q_name1_text, q_name2_text = sub_node.text.split()
+        self.assertEqual(qname1, text_to_qname(q_name1_text, sub_node.nsmap))
+        self.assertEqual(qname2, text_to_qname(q_name2_text, sub_node.nsmap))
+
+    def test_update_xml_value_with_mandatory_check(self):
+        """Test when the value is None and the property is mandatory."""
+        self.property._is_optional = False
+        setattr(self.instance, self.property._local_var_name, None)
+        with self.assertRaises(ValueError, msg=f'mandatory value {self.sub_element_name} missing'):
+            self.property.update_xml_value(self.instance, self.node)
+
+    def test_update_xml_value_overwrites_existing(self):
+        """Test that the method overwrites existing sub-element text."""
+        existing_sub_node = etree.SubElement(self.node, self.sub_element_name.localname, nsmap=self.node.nsmap)
+        existing_sub_node.text = docname_from_qname(
+            utils.random_qname(namespace=self.sub_element_name.text),
+            ns_map=self.node.nsmap,
+        )
+        qname = utils.random_qname(namespace=self.sub_element_name.text)
+        setattr(self.instance, self.property._local_var_name, [qname])
+        self.property.update_xml_value(self.instance, self.node)
+        sub_node = self.node.find(self.sub_element_name.text, namespaces=self.node.nsmap)
+        self.assertEqual(sub_node.text, docname_from_qname(qname, ns_map=self.node.nsmap))

```

Comment:
maybe it is more readable / understandable if doing something like this:

```
existing_sub_node = etree.SubElement(self.node, self.sub_element_name)
existing_sub_node.text = existing_sub_node.prefix + ":random_local_name"
qname = utils.random_qname()
setattr(self.instance, self.property._local_var_name, [qname])
self.property.update_xml_value(self.instance, self.node)
sub_node = self.node.find(self.sub_element_name.text)
self.assertEqual(sub_node.text, docname_from_qname(qname, ns_map=sub_node.nsmap))
```

Reply:
done